pkg/openstack/common.go

```diff
@@ -599,7 +619,6 @@ func (ed *EndpointDetail) CreateRoute(
 				return ctrlResult, nil
 			}
 		}
-
 		// create default TLS route override
```

Comment:
when `hasCertInOverrideSpec(ed.Route.OverrideSpec)` is true, I think we should also run the condition in L632 to get the `DestinationCaCertificate` if the user has not provided it and set the Termination to Reencrypt, or do we expect the end user to provide all this information as part of the `route.OverrideSpec` . It probably won't hurt to do it, since if something else is provided via the OverrideSpec by the user, it will have precedence.

Reply:
I'm not sure I understood 100%. But basically we can just add the `hasCertInOverrideSpec(ed.Route.OverrideSpec)` to:

`// for internal TLS (TLSE) use routev1.TLSTerminationReencrypt
		if ed.Service.TLS.Enabled && (ed.Service.TLS.SecretName != nil || hasCertInOverrideSpec(ed.Route.OverrideSpec)) {
......`

To set the termination to reencrypt, if the user doesn't directly specify it.

Reply:
yes thats what I meant. I think that's needed so in case the user did not set the reencrypt and added the dst ca cert to the override, it would be added by the operator.